src/classes/speciessite.cpp

```diff
@@ -272,240 +276,212 @@ std::vector<int> SpeciesSite::yAxisAtomIndices() const
+                    std::find(dynamicAtomTypes_.begin(), dynamicAtomTypes_.end(), i.atomType()) != dynamicAtomTypes_.end())
+                    sitesOriginAtomsIndices_.push_back({i.index()});
+            }
+            return true;
+        }
+        case (SiteType::Fragment):
+        {
+            std::vector<std::vector<int>> matchedIndices;
+            for (auto &i : parent_->atoms())
+            {
+                if (fragment_.matches(&i))
+                {
+                    // Determine the path of matched atoms - i.e. the atoms in the fragment.
+                    auto matchedGroup = fragment_.matchedPath(&i);
+                    auto matchedAtoms = matchedGroup.set();
+
+                    // Create vector of indices of the matched atoms.
+                    std::vector<int> matchedAtomIndices(matchedAtoms.size());
+                    std::transform(matchedAtoms.begin(), matchedAtoms.end(), matchedAtomIndices.begin(),
+                                   [](const auto &atom) { return atom->index(); });
+
+                    // Check if the fragment we have found is unique.
+                    std::sort(matchedAtomIndices.begin(), matchedAtomIndices.end());
+                    if (std::find(matchedIndices.begin(), matchedIndices.end(), matchedAtomIndices) != matchedIndices.end())
+                        continue;
+
+                    // If it's unique, remember it and proceed.
+                    matchedIndices.push_back(std::move(matchedAtomIndices));
+
+                    auto identifiers = matchedGroup.identifiers();
+
+                    // Determine origin atoms
+                    sitesOriginAtomsIndices_.emplace_back(identifiers["origin"].size());
+                    std::transform(identifiers["origin"].begin(), identifiers["origin"].end(),
+                                   sitesOriginAtomsIndices_.back().begin(), [](const auto &atom) { return atom->index(); });
+                    if (hasAxes())
+                    {
+                        // Determine x axis atoms.
+                        sitesXAxisAtomsIndices_.emplace_back(identifiers["x"].size());
+                        std::transform(identifiers["x"].begin(), identifiers["x"].end(), sitesXAxisAtomsIndices_.back().begin(),
+                                       [](const auto &atom) { return atom->index(); });
+                        // Determine y axis atoms.
+                        sitesYAxisAtomsIndices_.emplace_back(identifiers["y"].size());
+                        std::transform(identifiers["y"].begin(), identifiers["y"].end(), sitesYAxisAtomsIndices_.back().begin(),
+                                       [](const auto &atom) { return atom->index(); });
+                    }
+                }
+            }
+            return true;
+        }
+        default:
+        {
+            Messenger::print("Can't generate unique sites for site '{}'.", name());
+            return false;
+        }
```

Comment:
This is probably important so we should show a proper error:
```suggestion
            return Messenger::error("Can't generate unique sites for site '{}'.", name());
```

Reply:
Agreed.